src/core/thread/mle.cpp

```diff
@@ -2379,29 +2386,17 @@ otError Mle::SendAnnounce(uint8_t aChannel, bool aOrphanAnnounce, const Ip6::Add
 
 otError Mle::SendOrphanAnnounce(void)
 {
-    otError                            error = OT_ERROR_NONE;
-    const MeshCoP::ChannelMaskBaseTlv *channelMaskTlv;
-    const MeshCoP::ChannelMaskEntry *  channelMaskEntry;
-    MeshCoP::Dataset                   dataset(MeshCoP::Tlv::kActiveTimestamp);
-
-    SuccessOrExit(error = GetNetif().GetActiveDataset().Get(dataset));
-
-    channelMaskTlv = static_cast<const MeshCoP::ChannelMaskBaseTlv *>(dataset.Get(MeshCoP::Tlv::kChannelMask));
-    VerifyOrExit(channelMaskTlv != NULL, error = OT_ERROR_NOT_FOUND);
-
-    channelMaskEntry = channelMaskTlv->GetMaskEntry(OT_RADIO_CHANNEL_PAGE);
-    VerifyOrExit(channelMaskEntry != NULL, error = OT_ERROR_NOT_FOUND);
+    otError          error;
+    Mac::ChannelMask channelMask;
 
-    VerifyOrExit(mAnnounceChannel <= OT_RADIO_CHANNEL_MAX, error = OT_ERROR_NOT_FOUND);
+    SuccessOrExit(error = GetActiveDatasetChannelMask(channelMask));
 
-    while (!channelMaskEntry->IsChannelSet(mAnnounceChannel))
+    if (channelMask.GetNextChannel(mAnnounceChannel) != OT_ERROR_NONE)
     {
-        mAnnounceChannel++;
-        VerifyOrExit(mAnnounceChannel <= OT_RADIO_CHANNEL_MAX, error = OT_ERROR_NOT_FOUND);
+        ExitNow(error = OT_ERROR_NOT_FOUND);
```

Comment:
suggestion: How about simplifying this with `SuccessOrExit()`? (`GetNextChannel()` returns `OT_ERROR_NOT_FOUND` on failure). 
```
SuccessOrExit(error = channelMask.GetNextChannel(mAnnounceChannel));
```
in place of 
```
if (channelMask.GetNextChannel(mAnnounceChannel) != OT_ERROR_NONE)
{ 
    ExitNow(error = OT_ERROR_NOT_FOUND);
}
```

Reply:
Changed.